### ISSUE TITLE
command '/usr/local/cuda-10.1/bin/nvcc' failed with exit status 1

### ISSUE THREAD

**Issue by shuangyichen** [facebookresearch/detectron2#1495]
If you do not know the root cause of the problem, and wish someone to help you, please
post according to this template:

## Instructions To Reproduce the Issue:

1. what changes you made (`git diff`) or what code you wrote
```
Didnt change any.
```
2. what exact command you run:python -m pip install -e detectron2
3. what you observed (including __full logs__):
```
Obtaining file:///root/detectron2
Requirement already satisfied: termcolor>=1.1 in /usr/local/lib/python3.6/site-packages (from detectron2==0.1.3) (1.1.0)
Requirement already satisfied: Pillow in /usr/local/lib64/python3.6/site-packages (from detectron2==0.1.3) (6.1.0)
Requirement already satisfied: yacs>=0.1.6 in /usr/local/lib/python3.6/site-packages (from detectron2==0.1.3) (0.1.7)
Requirement already satisfied: tabulate in /usr/local/lib/python3.6/site-packages (from detectron2==0.1.3) (0.8.7)
Requirement already satisfied: cloudpickle in /usr/local/lib/python3.6/site-packages (from detectron2==0.1.3) (1.4.1)
Requirement already satisfied: matplotlib in /usr/local/lib64/python3.6/site-packages (from detectron2==0.1.3) (3.2.1)
Requirement already satisfied: mock in /usr/local/lib/python3.6/site-packages (from detectron2==0.1.3) (4.0.2)
Requirement already satisfied: tqdm>4.29.0 in /usr/local/lib/python3.6/site-packages (from detectron2==0.1.3) (4.46.0)
Requirement already satisfied: tensorboard in /usr/local/lib/python3.6/site-packages (from detectron2==0.1.3) (1.14.0)
Requirement already satisfied: fvcore>=0.1.1 in /usr/local/lib/python3.6/site-packages (from detectron2==0.1.3) (0.1.1.post200513)
Requirement already satisfied: future in /usr/local/lib/python3.6/site-packages (from detectron2==0.1.3) (0.18.2)
Requirement already satisfied: pydot in /usr/local/lib/python3.6/site-packages (from detectron2==0.1.3) (1.4.1)
Requirement already satisfied: PyYAML in /usr/local/lib64/python3.6/site-packages (from yacs>=0.1.6->detectron2==0.1.3) (5.3.1)
Requirement already satisfied: numpy>=1.11 in /usr/local/lib64/python3.6/site-packages (from matplotlib->detectron2==0.1.3) (1.15.0)
Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib64/python3.6/site-packages (from matplotlib->detectron2==0.1.3) (1.2.0)
Requirement already satisfied: python-dateutil>=2.1 in /usr/local/lib/python3.6/site-packages (from matplotlib->detectron2==0.1.3) (2.8.1)
Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 in /usr/local/lib/python3.6/site-packages (from matplotlib->detectron2==0.1.3) (2.4.7)
Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.6/site-packages (from matplotlib->detectron2==0.1.3) (0.10.0)
Requirement already satisfied: grpcio>=1.6.3 in /usr/local/lib64/python3.6/site-packages (from tensorboard->detectron2==0.1.3) (1.27.2)
Requirement already satisfied: absl-py>=0.4 in /usr/local/lib/python3.6/site-packages (from tensorboard->detectron2==0.1.3) (0.9.0)
Requirement already satisfied: six>=1.10.0 in /usr/local/lib/python3.6/site-packages (from tensorboard->detectron2==0.1.3) (1.14.0)
Requirement already satisfied: werkzeug>=0.11.15 in /usr/local/lib/python3.6/site-packages (from tensorboard->detectron2==0.1.3) (1.0.0)
Requirement already satisfied: setuptools>=41.0.0 in /usr/local/lib/python3.6/site-packages (from tensorboard->detectron2==0.1.3) (46.2.0)
Requirement already satisfied: protobuf>=3.6.0 in /usr/local/lib64/python3.6/site-packages (from tensorboard->detectron2==0.1.3) (3.11.3)
Requirement already satisfied: wheel>=0.26; python_version >= "3" in /usr/local/lib/python3.6/site-packages (from tensorboard->detectron2==0.1.3) (0.34.2)
Requirement already satisfied: markdown>=2.6.8 in /usr/local/lib/python3.6/site-packages (from tensorboard->detectron2==0.1.3) (3.2.1)
Requirement already satisfied: portalocker in /usr/local/lib/python3.6/site-packages (from fvcore>=0.1.1->detectron2==0.1.3) (1.7.0)
Installing collected packages: detectron2
  Attempting uninstall: detectron2
    Found existing installation: detectron2 0.1.3
    Can't uninstall 'detectron2'. No files were found to uninstall.
  Running setup.py develop for detectron2
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/root/detectron2/setup.py'"'"'; __file__='"'"'/root/detectron2/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps
         cwd: /root/detectron2/
    Complete output (48 lines):
    which: no hipcc in (/usr/bin/bazel/output:/root/anaconda3/bin:/usr/local/cuda-10.1/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/TensorRT-7.0.0.11:/root/bin)
    running develop
    running egg_info
    writing detectron2.egg-info/PKG-INFO
    writing dependency_links to detectron2.egg-info/dependency_links.txt
    writing requirements to detectron2.egg-info/requires.txt
    writing top-level names to detectron2.egg-info/top_level.txt
    reading manifest file 'detectron2.egg-info/SOURCES.txt'
    writing manifest file 'detectron2.egg-info/SOURCES.txt'
    running build_ext
    building 'detectron2._C' extension
    creating build
    creating build/temp.linux-x86_64-3.6
    creating build/temp.linux-x86_64-3.6/root
    creating build/temp.linux-x86_64-3.6/root/detectron2
    creating build/temp.linux-x86_64-3.6/root/detectron2/detectron2
    creating build/temp.linux-x86_64-3.6/root/detectron2/detectron2/layers
    creating build/temp.linux-x86_64-3.6/root/detectron2/detectron2/layers/csrc
    creating build/temp.linux-x86_64-3.6/root/detectron2/detectron2/layers/csrc/ROIAlign
    creating build/temp.linux-x86_64-3.6/root/detectron2/detectron2/layers/csrc/ROIAlignRotated
    creating build/temp.linux-x86_64-3.6/root/detectron2/detectron2/layers/csrc/box_iou_rotated
    creating build/temp.linux-x86_64-3.6/root/detectron2/detectron2/layers/csrc/nms_rotated
    creating build/temp.linux-x86_64-3.6/root/detectron2/detectron2/layers/csrc/deformable
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -DWITH_CUDA -I/root/detectron2/detectron2/layers/csrc -I/usr/local/lib64/python3.6/site-packages/torch/include -I/usr/local/lib64/python3.6/site-packages/torch/include/torch/csrc/api/include -I/usr/local/lib64/python3.6/site-packages/torch/include/TH -I/usr/local/lib64/python3.6/site-packages/torch/include/THC -I/usr/local/cuda-10.1/include -I/usr/include/python3.6m -c /root/detectron2/detectron2/layers/csrc/vision.cpp -o build/temp.linux-x86_64-3.6/root/detectron2/detectron2/layers/csrc/vision.o -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_C -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -DWITH_CUDA -I/root/detectron2/detectron2/layers/csrc -I/usr/local/lib64/python3.6/site-packages/torch/include -I/usr/local/lib64/python3.6/site-packages/torch/include/torch/csrc/api/include -I/usr/local/lib64/python3.6/site-packages/torch/include/TH -I/usr/local/lib64/python3.6/site-packages/torch/include/THC -I/usr/local/cuda-10.1/include -I/usr/include/python3.6m -c /root/detectron2/detectron2/layers/csrc/ROIAlign/ROIAlign_cpu.cpp -o build/temp.linux-x86_64-3.6/root/detectron2/detectron2/layers/csrc/ROIAlign/ROIAlign_cpu.o -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_C -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -DWITH_CUDA -I/root/detectron2/detectron2/layers/csrc -I/usr/local/lib64/python3.6/site-packages/torch/include -I/usr/local/lib64/python3.6/site-packages/torch/include/torch/csrc/api/include -I/usr/local/lib64/python3.6/site-packages/torch/include/TH -I/usr/local/lib64/python3.6/site-packages/torch/include/THC -I/usr/local/cuda-10.1/include -I/usr/include/python3.6m -c /root/detectron2/detectron2/layers/csrc/ROIAlignRotated/ROIAlignRotated_cpu.cpp -o build/temp.linux-x86_64-3.6/root/detectron2/detectron2/layers/csrc/ROIAlignRotated/ROIAlignRotated_cpu.o -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_C -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -DWITH_CUDA -I/root/detectron2/detectron2/layers/csrc -I/usr/local/lib64/python3.6/site-packages/torch/include -I/usr/local/lib64/python3.6/site-packages/torch/include/torch/csrc/api/include -I/usr/local/lib64/python3.6/site-packages/torch/include/TH -I/usr/local/lib64/python3.6/site-packages/torch/include/THC -I/usr/local/cuda-10.1/include -I/usr/include/python3.6m -c /root/detectron2/detectron2/layers/csrc/box_iou_rotated/box_iou_rotated_cpu.cpp -o build/temp.linux-x86_64-3.6/root/detectron2/detectron2/layers/csrc/box_iou_rotated/box_iou_rotated_cpu.o -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_C -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -DWITH_CUDA -I/root/detectron2/detectron2/layers/csrc -I/usr/local/lib64/python3.6/site-packages/torch/include -I/usr/local/lib64/python3.6/site-packages/torch/include/torch/csrc/api/include -I/usr/local/lib64/python3.6/site-packages/torch/include/TH -I/usr/local/lib64/python3.6/site-packages/torch/include/THC -I/usr/local/cuda-10.1/include -I/usr/include/python3.6m -c /root/detectron2/detectron2/layers/csrc/nms_rotated/nms_rotated_cpu.cpp -o build/temp.linux-x86_64-3.6/root/detectron2/detectron2/layers/csrc/nms_rotated/nms_rotated_cpu.o -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_C -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++14
    /usr/local/cuda-10.1/bin/nvcc -DWITH_CUDA -I/root/detectron2/detectron2/layers/csrc -I/usr/local/lib64/python3.6/site-packages/torch/include -I/usr/local/lib64/python3.6/site-packages/torch/include/torch/csrc/api/include -I/usr/local/lib64/python3.6/site-packages/torch/include/TH -I/usr/local/lib64/python3.6/site-packages/torch/include/THC -I/usr/local/cuda-10.1/include -I/usr/include/python3.6m -c /root/detectron2/detectron2/layers/csrc/ROIAlign/ROIAlign_cuda.cu -o build/temp.linux-x86_64-3.6/root/detectron2/detectron2/layers/csrc/ROIAlign/ROIAlign_cuda.o -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr --compiler-options '-fPIC' -DCUDA_HAS_FP16=1 -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_C -D_GLIBCXX_USE_CXX11_ABI=0 -gencode=arch=compute_70,code=sm_70 -std=c++14
    /usr/local/src/gcc-8.3.0/include/c++/8.3.0/bits/basic_string.tcc: In instantiation of ‘static std::basic_string<_CharT, _Traits, _Alloc>::_Rep* std::basic_string<_CharT, _Traits, _Alloc>::_Rep::_S_create(std::basic_string<_CharT, _Traits, _Alloc>::size_type, std::basic_string<_CharT, _Traits, _Alloc>::size_type, const _Alloc&) [with _CharT = char16_t; _Traits = std::char_traits<char16_t>; _Alloc = std::allocator<char16_t>; std::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’:
    /usr/local/src/gcc-8.3.0/include/c++/8.3.0/bits/basic_string.tcc:578:28:   required from ‘static _CharT* std::basic_string<_CharT, _Traits, _Alloc>::_S_construct(_InIterator, _InIterator, const _Alloc&, std::forward_iterator_tag) [with _FwdIterator = const char16_t*; _CharT = char16_t; _Traits = std::char_traits<char16_t>; _Alloc = std::allocator<char16_t>]’
    /usr/local/src/gcc-8.3.0/include/c++/8.3.0/bits/basic_string.h:5052:20:   required from ‘static _CharT* std::basic_string<_CharT, _Traits, _Alloc>::_S_construct_aux(_InIterator, _InIterator, const _Alloc&, std::__false_type) [with _InIterator = const char16_t*; _CharT = char16_t; _Traits = std::char_traits<char16_t>; _Alloc = std::allocator<char16_t>]’
    /usr/local/src/gcc-8.3.0/include/c++/8.3.0/bits/basic_string.h:5073:24:   required from ‘static _CharT* std::basic_string<_CharT, _Traits, _Alloc>::_S_construct(_InIterator, _InIterator, const _Alloc&) [with _InIterator = const char16_t*; _CharT = char16_t; _Traits = std::char_traits<char16_t>; _Alloc = std::allocator<char16_t>]’
    /usr/local/src/gcc-8.3.0/include/c++/8.3.0/bits/basic_string.tcc:656:134:   required from ‘std::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _CharT*, std::basic_string<_CharT, _Traits, _Alloc>::size_type, const _Alloc&) [with _CharT = char16_t; _Traits = std::char_traits<char16_t>; _Alloc = std::allocator<char16_t>; std::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’
    /usr/local/src/gcc-8.3.0/include/c++/8.3.0/bits/basic_string.h:6725:95:   required from here
    /usr/local/src/gcc-8.3.0/include/c++/8.3.0/bits/basic_string.tcc:1067:1: error: cannot call member function ‘void std::basic_string<_CharT, _Traits, _Alloc>::_Rep::_M_set_sharable() [with _CharT = char16_t; _Traits = std::char_traits<char16_t>; _Alloc = std::allocator<char16_t>]’ without object
           __p->_M_set_sharable();
     ^     ~~~~~~~~~
    /usr/local/src/gcc-8.3.0/include/c++/8.3.0/bits/basic_string.tcc: In instantiation of ‘static std::basic_string<_CharT, _Traits, _Alloc>::_Rep* std::basic_string<_CharT, _Traits, _Alloc>::_Rep::_S_create(std::basic_string<_CharT, _Traits, _Alloc>::size_type, std::basic_string<_CharT, _Traits, _Alloc>::size_type, const _Alloc&) [with _CharT = char32_t; _Traits = std::char_traits<char32_t>; _Alloc = std::allocator<char32_t>; std::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’:
    /usr/local/src/gcc-8.3.0/include/c++/8.3.0/bits/basic_string.tcc:578:28:   required from ‘static _CharT* std::basic_string<_CharT, _Traits, _Alloc>::_S_construct(_InIterator, _InIterator, const _Alloc&, std::forward_iterator_tag) [with _FwdIterator = const char32_t*; _CharT = char32_t; _Traits = std::char_traits<char32_t>; _Alloc = std::allocator<char32_t>]’
    /usr/local/src/gcc-8.3.0/include/c++/8.3.0/bits/basic_string.h:5052:20:   required from ‘static _CharT* std::basic_string<_CharT, _Traits, _Alloc>::_S_construct_aux(_InIterator, _InIterator, const _Alloc&, std::__false_type) [with _InIterator = const char32_t*; _CharT = char32_t; _Traits = std::char_traits<char32_t>; _Alloc = std::allocator<char32_t>]’
    /usr/local/src/gcc-8.3.0/include/c++/8.3.0/bits/basic_string.h:5073:24:   required from ‘static _CharT* std::basic_string<_CharT, _Traits, _Alloc>::_S_construct(_InIterator, _InIterator, const _Alloc&) [with _InIterator = const char32_t*; _CharT = char32_t; _Traits = std::char_traits<char32_t>; _Alloc = std::allocator<char32_t>]’
    /usr/local/src/gcc-8.3.0/include/c++/8.3.0/bits/basic_string.tcc:656:134:   required from ‘std::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _CharT*, std::basic_string<_CharT, _Traits, _Alloc>::size_type, const _Alloc&) [with _CharT = char32_t; _Traits = std::char_traits<char32_t>; _Alloc = std::allocator<char32_t>; std::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’
    /usr/local/src/gcc-8.3.0/include/c++/8.3.0/bits/basic_string.h:6730:95:   required from here
    /usr/local/src/gcc-8.3.0/include/c++/8.3.0/bits/basic_string.tcc:1067:1: error: cannot call member function ‘void std::basic_string<_CharT, _Traits, _Alloc>::_Rep::_M_set_sharable() [with _CharT = char32_t; _Traits = std::char_traits<char32_t>; _Alloc = std::allocator<char32_t>]’ without object
    /usr/local/lib64/python3.6/site-packages/torch/utils/cpp_extension.py:304: UserWarning: Attempted to use ninja as the BuildExtension backend but we could not find ninja.. Falling back to using the slow distutils backend.
      warnings.warn(msg.format('we could not find ninja.'))
    error: command '/usr/local/cuda-10.1/bin/nvcc' failed with exit status 1
    ----------------------------------------
  ERROR: Can't roll back detectron2; was not uninstalled
ERROR: Command errored out with exit status 1: /usr/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/root/detectron2/setup.py'"'"'; __file__='"'"'/root/detectron2/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps Check the logs for full command output.

```
4. please simplify the steps as much as possible so they do not require additional resources to
	 run, such as a private dataset.

## Expected behavior:

If there are no obvious error in "what you observed" provided above,
please tell us the expected behavior.

If you expect the model to converge / work better, note that we do not give suggestions
on how to train a new model.
Only in one of the two conditions we will help with it:
(1) You're unable to reproduce the results in detectron2 model zoo.
(2) It indicates a detectron2 bug.

## Environment:

Provide your environment information using the following command:
```
which: no hipcc in (/usr/bin/bazel/output:/root/anaconda3/bin:/usr/local/cuda-10.1/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/TensorRT-7.0.0.11:/root/bin)
----------------------  ------------------------------------------------------------------------------
sys.platform            linux
Python                  3.6.8 (default, Apr  2 2020, 13:34:55) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
numpy                   1.15.0
detectron2              0.1.3 @/root/detectron2/detectron2
detectron2._C           failed to import
Compiler                c++ (GCC) 4.8.5 20150623 (Red Hat 4.8.5-39)
CUDA compiler           Cuda compilation tools, release 10.1, V10.1.105
DETECTRON2_ENV_MODULE   <not set>
PyTorch                 1.5.0+cu101 @/usr/local/lib64/python3.6/site-packages/torch
PyTorch debug build     False
CUDA available          True
GPU 0                   Tesla V100-PCIE-16GB
CUDA_HOME               /usr/local/cuda-10.1
Pillow                  6.1.0
torchvision             0.6.0+cu101 @/usr/local/lib64/python3.6/site-packages/torchvision
torchvision arch flags  sm_35, sm_50, sm_60, sm_70, sm_75
fvcore                  0.1.1.post200513
cv2                     4.2.0
----------------------  ------------------------------------------------------------------------------
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2019.0.5 Product Build 20190808 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v0.21.1 (Git Hash 7d2fd500bc78936d1d648ca713b901012f470dbc)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: NO AVX
  - CUDA Runtime 10.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.3
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_INTERNAL_THREADPOOL_IMPL -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF, 

```

If your issue looks like an installation issue / environment issue,
please first try to solve it yourself with the instructions in
https://detectron2.readthedocs.io/tutorials/install.html#common-installation-issues

**Comment by ppwwyyxx** [comment]
Compiler is too old according to https://detectron2.readthedocs.io/tutorials/install.html

**Comment by shuangyichen** [comment]
Upgrade compiler to gcc7, still failed

sys.platform            linux
Python                  3.6.8 (default, Apr  2 2020, 13:34:55) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
numpy                   1.15.0
detectron2              0.1.3 @/root/detectron2/detectron2
detectron2._C           failed to import
Compiler                c++ (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
CUDA compiler           Cuda compilation tools, release 10.1, V10.1.105
DETECTRON2_ENV_MODULE   <not set>
PyTorch                 1.5.0+cu101 @/usr/local/lib64/python3.6/site-packages/torch
PyTorch debug build     False
CUDA available          True
GPU 0                   Tesla V100-PCIE-16GB
CUDA_HOME               /usr/local/cuda-10.1
Pillow                  6.1.0
torchvision             0.6.0+cu101 @/usr/local/lib64/python3.6/site-packages/torchvision
torchvision arch flags  sm_35, sm_50, sm_60, sm_70, sm_75
fvcore                  0.1.1.post200513
cv2                     4.2.0
----------------------  ------------------------------------------------------------------------------
PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2019.0.5 Product Build 20190808 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v0.21.1 (Git Hash 7d2fd500bc78936d1d648ca713b901012f470dbc)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: NO AVX
  - CUDA Runtime 10.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.3
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_INTERNAL_THREADPOOL_IMPL -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_STATIC_DISPATCH=OFF,

**Comment by ppwwyyxx** [comment]
Please provide full logs again.

**Comment by Yueying111** [comment]
I have met with the same problem, and I resolved it by uninstalling all cudas installed on my computer( I tried a few times, and there were too many versions of cuda on my computer) and reinstall it and everything works. Also don't forget to add path in your bashrc.  Try following this https://itectec.com/ubuntu/ubuntu-how-to-install-nvidia-and-cuda-drivers-into-ubuntu/
I'm really new in coding, hope the experience could help you a bit.
